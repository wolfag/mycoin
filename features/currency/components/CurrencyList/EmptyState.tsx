import { getTestId } from '@/shared/utils/getTestId';
import AntDesign from '@expo/vector-icons/AntDesign';
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

const EmptyState = () => {
  return (
    <View style={styles.container} {...getTestId('EmptyState')}>
      <AntDesign name='frown' size={60} color='grey' />
      <Text style={styles.firstText}>No Results</Text>
      <Text style={styles.secondText}>{`Try "MCO"`}</Text>
    </View>
  );
};

export default EmptyState;

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
  },
  firstText: {
    marginTop: 10,
  },
  secondText: {
    color: 'grey',
    marginTop: 5,
  },
});
