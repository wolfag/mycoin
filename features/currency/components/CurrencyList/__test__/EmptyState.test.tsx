import { render, waitFor } from '@testing-library/react-native';
import EmptyState from '../EmptyState';

describe('EmptyState', () => {
  it('renders UI', async () => {
    const { toJSON } = render(<EmptyState />);

    await waitFor(() => {
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
