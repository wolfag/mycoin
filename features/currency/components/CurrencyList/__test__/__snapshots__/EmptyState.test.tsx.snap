// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState renders UI 1`] = `
<View
  style={
    {
      "alignItems": "center",
    }
  }
  testID="EmptyState"
>
  <AntDesign
    color="grey"
    name="frown"
    size={60}
  />
  <Text
    style={
      {
        "marginTop": 10,
      }
    }
  >
    No Results
  </Text>
  <Text
    style={
      {
        "color": "grey",
        "marginTop": 5,
      }
    }
  >
    Try "MCO"
  </Text>
</View>
`;
